encode.go

```diff
@@ -10,107 +10,77 @@ import (
+		// Scalar & Vector of boolean, int variants, float variant, and bytes.
+		// Note: UUID is back by byte array of length 16
 		binary.Write(dbuf, order, data.Data)
-	// Vector types backed by any variant of boolean, int, float, bytes
-	case KB, UU, KG, KH, KI, KJ, KE, KF, KM, KD, KZ, KN, KU, KV, KT:
-		binary.Write(dbuf, order, data.Type)
-		binary.Write(dbuf, order, data.Attr)
-		binary.Write(dbuf, order, int32(reflect.ValueOf(data.Data).Len()))
-		binary.Write(dbuf, order, data.Data)
-	// String
-	case KC:
-		toSend := data.Data.(string)
-
-		binary.Write(dbuf, order, data.Type)
-		binary.Write(dbuf, order, data.Attr)
-		binary.Write(dbuf, order, int32(len(toSend)))
-		binary.Write(dbuf, order, []byte(toSend))
-	// Scalar symbol
-	case -KS:
-		toSend := data.Data.(string)
-
-		binary.Write(dbuf, order, data.Type)
-		binary.Write(dbuf, order, []byte(toSend))
-		binary.Write(dbuf, order, byte(0))
-	// Vector symbol
-	case KS:
-		toSend := data.Data.([]string)
-
-		binary.Write(dbuf, order, data.Type)
-		binary.Write(dbuf, order, data.Attr)
-		binary.Write(dbuf, order, int32(len(toSend)))
-		for i := 0; i < len(toSend); i++ {
-			binary.Write(dbuf, order, []byte(toSend[i]))
-			binary.Write(dbuf, order, byte(0))
+	case KC: // String
+		dbuf.WriteString(data.Data.(string))
+	case -KS: // Scalar symbol
+		dbuf.WriteString(data.Data.(string))
+		binary.Write(dbuf, order, byte(0)) // Null terminator
+	case KS: // Vector symbol
+		for _, toSend := range data.Data.([]string) {
+			dbuf.WriteString(toSend)
+			binary.Write(dbuf, order, byte(0)) // Null terminator
 		}
-	case -KP:
-		toSend := data.Data.(time.Time)
-
-		binary.Write(dbuf, order, data.Type)
-		binary.Write(dbuf, order, toSend.Sub(qEpoch))
-	case KP:
-		binary.Write(dbuf, order, data.Type)
-		binary.Write(dbuf, order, data.Attr)
-		binary.Write(dbuf, order, int32(reflect.ValueOf(data.Data).Len()))
-		tosend := data.Data.([]time.Time)
-		for _, ts := range tosend {
-			binary.Write(dbuf, order, ts.Sub(qEpoch))
+	case -KP: // Scalar timestamp
+		binary.Write(dbuf, order, data.Data.(time.Time).Sub(qEpoch))
+	case KP: // Vector timestamp
+		for _, toSend := range data.Data.([]time.Time) {
+			binary.Write(dbuf, order, toSend.Sub(qEpoch))
 		}
-	case XD:
-		tosend := data.Data.(Dict)
-		binary.Write(dbuf, order, XD)
-		err = writeData(dbuf, order, tosend.Key)
+	case XD: // Dictionary
+		toSend := data.Data.(Dict)
+		err := writeData(dbuf, order, toSend.Key)
 		if err != nil {
 			return err
 		}
-		err = writeData(dbuf, order, tosend.Value)
+		err = writeData(dbuf, order, toSend.Value)
 		if err != nil {
 			return err
 		}
-	case XT:
-		tosend := data.Data.(Table)
-		binary.Write(dbuf, order, XT)
-		binary.Write(dbuf, order, data.Attr)
-		err = writeData(dbuf, order, &K{XD, NONE, Dict{&K{KS, NONE, tosend.Columns}, &K{K0, NONE, tosend.Data}}})
+	case XT: // Table
+		toSend := data.Data.(Table)
+		err := writeData(dbuf, order, NewDict(SymbolV(toSend.Columns), Enlist(toSend.Data...)))
```

Comment:
How did you verify that table is being encoded the same way as before? I don't see any unit tests.

Reply:
Never mind they're there.